apps/sms/services/js/moz_settings/moz_settings_shim.js

```diff
@@ -0,0 +1,74 @@
+/* -*- Mode: js; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+/* global bridge */
+
+/* exported MozSettingsShim */
+
+(function(exports) {
+'use strict';
+
+/**
+ * Name of the service for mozSettings API shim.
+ * @type {string}
+ */
+const SERVICE_NAME = 'moz-settings-shim';
+
+/**
+ * Array of method names that need to be exposed for API shim.
+ * @type {Array.<string>}
+ */
+const METHODS = ['get', 'set', 'addObserver'];
+
+var mozSettings = null;
+
+var service = null;
+
+var MozSettingsShim = {
+  init(settings) {
+    if (!settings) {
+      return;
+    }
+
+    mozSettings = settings;
+    service = bridge.service(SERVICE_NAME);
+
+    METHODS.forEach((shimMethod) => {
+      service.method(shimMethod, this[shimMethod]);
+    });
+  },
+
+  /* Methods */
+
+  /**
+   * Shim for mozSettings.createLock().get API.
+   * @param {String} key Key name for Settings.
+   * @return {Promise} Promise that return the settings value from settings DB.
+   */
+  get(key) {
+    return mozSettings.createLock().get(key);
+  },
+
+  /**
+   * Shim for mozSettings.createLock().set API.
+   * @param {Object} setting Settings key and value for .
+   * @return {Promise} Promise that return saved result.
+   */
+  set(setting) {
+    return mozSettings.createLock().set(setting);
+  },
+
+  /**
+   * Add the Observer and return the changed value via service broadcast.
+   * @param {String} key Key name for Settings.
+   */
+  addObserver(key) {
+    mozSettings.addObserver(key, (e) => {
+      service.broadcast(key, e.settingValue);
+    });
```

Comment:
I don't think we need the observer now; I think we just need to get values when we need them.

If we want an addObserver implementation this will need to be more complicated: add only one real observer, and be able to remove observers; because this file may be loaded only once for several clients.

Not sure we really need this complexity, what do you think ?